src/SqlMigrationTool.php

```diff
@@ -35,22 +37,29 @@ class SqlMigrationTool extends AbstractMigrationTool
      * @var bool
      */
     private $supportsMultiStatements;
+    
+    /**
+     *
+     * @var DbAdapter
+     */
+    private $adapter;
 
     /**
      * @param string $directory
      * @param \PDO $pdo
      * @param string $migrationTableName
      * @param string $filePattern
      */
-    public function __construct($directory, \PDO $pdo, $migrationTableName = 'migrations', $filePattern = '/\.sql$/i')
+    public function __construct($directory, \PDO $pdo, $migrationTableName = 'migrations', $filePattern = '/\.sql$/i', DbAdapter $adapter = null)
     {
         parent::__construct($directory, $filePattern);
         $this->pdo = $pdo;
         $this->migrationTableName = $migrationTableName;
+        $this->adapter = ($adapter) ? $adapter : new MySqlAdapter();
 
         $this->pdo->setAttribute(\PDO::ATTR_ERRMODE, \PDO::ERRMODE_EXCEPTION);
         $this->supportsMultiStatements = in_array($this->pdo->getAttribute(\PDO::ATTR_DRIVER_NAME), [
-            "mysql"
+           $this->adapter->getDbType()
```

Comment:
This part checks if `$this->pdo` is capable to run multi statement. You can append another driver name to this array. Your change means that every adapter supports it.

Reply:
Maybe I don't undenstand you last sence. If adapter returns string which is not in array that means that multi select are not supporter, isn't it?